pcdsdevices/mirror.py

```diff
@@ -519,6 +742,60 @@ class XOffsetMirrorBend(XOffsetMirror):
     # Tab config: show components
     tab_component_names = True
 
+    def calc_lightpath_state(
+        self,
+        x_up: float,
+        y_up: float,
+        pitch: float
+    ) -> LightpathState:
+        """
+        Special lightpath calculation for mr1k1_bend, which only
+        depends on y-range for insertion, rather than x-range
+
+        Parameters
+        ----------
+        x_up : float
+            x position, unused
+        y_up : float
+            y position, used to determine insertion
+        pitch : float
+            pitch position, unused
+
+        Returns
+        -------
+        LightpathState
+            current lightpath state of the device
+        """
+        try:
+            if np.shape(self.y_ranges) != (2, 2):
+                # improper ranges for insertion, fail
+                raise MirrorLogicError(
+                    'Provided x-ranges are the malformed. '
+                    f'got: {np.shape(self.x_ranges)}, expected (2,2)')
+
+            x_out, x_in = ((limit[0] < y_up) and (y_up < limit[1])
```

Comment:
This same loop is here a few times; you might consider a helper that does the logic and validation:

```python
def find_matching_range_indices(ranges, value) -> List[int]:
  if np.shape...
```